files/default/neutron-ha-tool.py

```diff
@@ -943,45 +970,43 @@ def list_alive_agents(agent_list, agent_type):
     return [agent for agent in agent_list
             if agent['agent_type'] == agent_type and
             agent['alive'] is True and
-            agent['admin_state_up'] is True]
+            agent['admin_state_up'] is True and
+            (not exclude_agent_mode or
+             agent['configurations']['agent_mode'] != exclude_agent_mode)]
 
 
-def target_agent_list(agent_list, agent_type, exclude_agent_host):
+def target_agent_list(agent_list, agent_type, exclude_agent):
     """
     Return a list of agents that are alive, excluding the one we want to
-    migrate from
+    migrate from and filtering out agents that have a different agent
+    mode
 
     :param agent_list: API response for list_agents()
     :param agent_type: used to filter the type of agent
-    :param exclude_agent_host: hostname of agent to exclude from the list
+    :param exclude_agent: agent to exclude from the list
 
     """
-    agent_info = [
-        i for i in agent_list
-        if i.get('host', None) == exclude_agent_host
-    ]
-    if not agent_info:
-        LOG.error("Cannot find agent %s information.", exclude_agent_host)
-        return []
-    agent_info = agent_info[0]
-    agent_mode = agent_info['configurations']['agent_mode']
-
+    agent_mode = exclude_agent['configurations']['agent_mode']
     return [agent for agent in agent_list
             if agent['agent_type'] == agent_type and
             agent['alive'] and
-            agent['host'] != exclude_agent_host and
+            agent['admin_state_up'] is True and
```

Comment:
Yes, it is a separate fix, but with the latest update completely replacing the target_agent_list with the list_alive_l3_agents, this fix is completely factored into the changeset, so it's impossible to detach from it.